metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -156,278 +163,419 @@ def ensure_field_level_settings_are_normalized(
         return values
 
 
-class _DatasetProfiler(BasicDatasetProfilerBase):
-    @classmethod
-    def _get_columns_to_profile(
-        cls,
-        dataset: Dataset,
-        dataset_name: str,
-        config: GEProfilingConfig,
-        report: SQLSourceReport,
-    ) -> List[str]:
-        if config.profile_table_level_only:
+def _convert_to_cardinality(
+    unique_count: Optional[int], pct_unique: Optional[float]
+) -> Optional[OrderedProfilerCardinality]:
+    # Logic adopted from Great Expectations.
+
+    cardinality = None
+    if unique_count is None or unique_count == 0 or pct_unique is None:
+        cardinality = OrderedProfilerCardinality.NONE
+    elif pct_unique == 1.0:
+        cardinality = OrderedProfilerCardinality.UNIQUE
+    elif pct_unique > 0.1:
+        cardinality = OrderedProfilerCardinality.VERY_MANY
+    elif pct_unique > 0.02:
+        cardinality = OrderedProfilerCardinality.MANY
+    else:
+        if unique_count == 1:
+            cardinality = OrderedProfilerCardinality.ONE
+        elif unique_count == 2:
+            cardinality = OrderedProfilerCardinality.TWO
+        elif unique_count < 60:
+            cardinality = OrderedProfilerCardinality.VERY_FEW
+        elif unique_count < 1000:
+            cardinality = OrderedProfilerCardinality.FEW
+        else:
+            cardinality = OrderedProfilerCardinality.MANY
+
+    return cardinality
+
+
+def _is_single_row_query_method(query: Any) -> bool:
+    SINGLE_ROW_QUERY_FILE = "great_expectations/dataset/sqlalchemy_dataset.py"
+    SINGLE_ROW_QUERY_METHODS = {
+        "get_row_count",
+        "get_column_min",
+        "get_column_max",
+        "get_column_mean",
+        "get_column_stdev",
+        "get_column_stdev",
+        "get_column_nonnull_count",
+        "get_column_unique_count",
+        # This actually returns two rows, not a single row, so we can't combine it with other queries.
+        # "get_column_median",
+    }
+    COLUMN_MAP_QUERY_METHOD = "inner_wrapper"
+    COLUMN_MAP_QUERY_SINGLE_ROW_COLUMNS = [
+        "element_count",
+        "null_count",
+        "unexpected_count",
+    ]
+
+    # We'll do this the inefficient way since the arrays are pretty small.
+    stack = traceback.extract_stack()
+    for frame in stack:
+        if not frame.filename.endswith(SINGLE_ROW_QUERY_FILE):
+            continue
+        if frame.name in SINGLE_ROW_QUERY_METHODS:
+            return True
+
+        if frame.name == COLUMN_MAP_QUERY_METHOD:
+            # Some column map expectations are single-row.
+            # We can disambiguate by checking the column names.
+            query_columns = query.columns
+            column_names = [column.name for column in query_columns]
+
+            if column_names == COLUMN_MAP_QUERY_SINGLE_ROW_COLUMNS:
```

Comment:
is this always equality or can `column_names` be a sub-set of `COLUMN_MAP_QUERY_SINGLE_ROW_COLUMNS`?

Reply:
We do want equality here - a subset would be too loose and might allow table columns named "element_count" or similar to slip through